pyface/action/action_manager.py

```diff
@@ -45,7 +45,7 @@ class ActionManager(HasTraits):
     enabled = Bool(True)
 
     #: All of the contribution groups in the manager.
-    groups = Property(List(Group))
+    groups = Property(List(Instance(Group)), observe='_groups.items')
```

Comment:
Not entirely unrelated - this PR introduces an `IActionManager` interface and this brings the `ActionManger` into agreement with that.  The `observe` is sort of orthogonal, but the advertised interface is `List(Instance(Group))` so it would be potentially surprising if the property didn't fire trait change events.